### ISSUE TITLE
Date.prototype.toLocaleString gives wrong result. (month: "long")

### ISSUE THREAD

**Issue by headcr4sh** [andyearnshaw/Intl.js#117]
```javascript
Date.prototype.toLocaleString = intl.__localeSensitiveProtos.Date.toLocaleString;
var s = new Date(1,1).toLocaleString('en', { month: 'long' });
// Expected result: s === "January", current result: s === "Jan"
```

I tried with different languages (en, de, es, fr,...)
Format "long" doesn't seem to work as expected when trying to format months.

**Comment by caridy** [comment]
@headcr4sh what version of the polyfill are you using?

**Comment by caridy** [comment]
repro case:

```
$ node
> require('./')
...
> Intl === IntlPolyfill
false
> new Date(1,1).toLocaleString('en', { month: 'long' })
'February'
> new Intl.DateTimeFormat('en', {month: 'long'}).format(new Date(1,1))
'February'
> new IntlPolyfill.DateTimeFormat('en', {month: 'long'}).format(new Date(1,1))
'Feb'
```

I will look into the details.

**Comment by ollym** [comment]
@caridy using master:

```javascript
new IntlPolyfill.DateTimeFormat('en', { year: 'numeric', month: 'short' }).format(new Date)
```

Returns

```
2015 M07
```

**Comment by caridy** [comment]
yeah, there is something wrong with the new data, I'm investigating.

**Comment by caridy** [comment]
@ollym are you sure that `new IntlPolyfill.DateTimeFormat('en', { year: 'numeric', month: 'short' }).format(new Date` is giving you `2015 M07`? I'm getting `Jul 2015`, but I do get that value when using `fr-FR`, which is also concerning.

**Comment by caridy** [comment]
Ok, I figure why `fr-FR` was behaving incorrecting, just like any other non-root locale, will commit that soon.

**Comment by gmunumel** [comment]
@caridy, great to hear that. I get the same issue.

**Comment by caridy** [comment]
@ollym @gmunumel master was updated to solve part of the problem in this issue.

```
> new IntlPolyfill.DateTimeFormat('en', { year: 'numeric', month: 'short' }).format(new Date())
'Jul 2015'
> new IntlPolyfill.DateTimeFormat('en-US', { year: 'numeric', month: 'short' }).format(new Date())
'Jul 2015'
> new IntlPolyfill.DateTimeFormat('fr', { year: 'numeric', month: 'short' }).format(new Date())
'juil. 2015'
> new IntlPolyfill.DateTimeFormat('fr-FR', { year: 'numeric', month: 'short' }).format(new Date())
'juil. 2015'
```

**Comment by vectart** [comment]
@caridy Yesterday I've started with master version and got the same problem.

``` javascript
Poly = IntlPolyfill

new Poly.DateTimeFormat('en', {month: 'long'}).format(new Date(2015, 1, 5))
// "Feb"
new Intl.DateTimeFormat('en', {month: 'long'}).format(new Date(2015, 1, 5))
// "February"

new Poly.DateTimeFormat('sv', {month: 'long'}).format(new Date(2015, 1, 5))
// "feb."
new Intl.DateTimeFormat('sv', {month: 'long'}).format(new Date(2015, 1, 5))
// "Februari"
```

**Comment by ollym** [comment]
@caridy Again, using 1.0.0 same issue

**Comment by caridy** [comment]
Right. The issue is that we don't have that data at the moment. When you look at the available data from CLDR, you will see that the only match for `en` and `sv` locales are very limited, they both happen to use the same month format, which uses `LLL` which correspond to the short month:

https://github.com/andyearnshaw/Intl.js/blob/master/locale-data/json/en.json#L43
https://github.com/andyearnshaw/Intl.js/blob/master/locale-data/json/sv.json#L43

I will follow up with CLDR folks, that's why this is still open. If you look in the data, only those format patterns with `MMMM` will display long month.

**Comment by vectart** [comment]
@caridy I don't really understand why you apply to CLDR guys. The Intl.js library is able to produce long month name with `day` option and outputs short month name without it. How is that related to CLDR?!

```js
new IntlPolyfill.DateTimeFormat('sv', {month: 'long'}).format(new Date(2015, 1, 5))
// "feb."
new IntlPolyfill.DateTimeFormat('sv', {month: 'long', day: 'numeric'}).format(new Date(2015, 1, 5))
// "5 februari"
```
http://jsbin.com/foxise/edit?html,console

**Comment by caridy** [comment]
@vectart we do have the metadata for the long months, we just don't have the formatter for it. You can look at the specifications to learn how we find the best match based on the configuration provided by the user.

**Comment by olegshilov** [comment]
hi, i have the same issue.
![screen shot 2015-10-07 at 19 55 12](https://cloud.githubusercontent.com/assets/930121/10353125/90de5170-6d5c-11e5-80f6-8571bba9f539.png)

**Comment by headcr4sh** [comment]
Any chance that this issue is going to be fixed soon(ish)?
From what I can see, there are a bunch of issues that are closely related to the cause of this issue...
... anything that can be done to help solve the problems?

**Comment by caridy** [comment]
Fixed by PR #146, shipped in intl@1.1.0

**Comment by headcr4sh** [comment]
@caridy  Great news. Thanks a lot! :+1:
